.github/workflows/report.yml

```diff
@@ -1,26 +1,40 @@
 name: 'Test Report'
 on:
   workflow_run:
-    workflows: ['Build App']                     # runs after CI workflow
+    workflows: ['Test and Build'] # runs after CI workflow
+    branches:
+      - dev
+      - main
     types:
       - completed
+
 permissions:
   contents: read
   actions: read
   checks: write
+
 jobs:
   report:
+    # Only run if the workflow_run succeeded AND it was triggered by a PR
+    if: >
+      github.event.workflow_run.conclusion == 'success' &&
+      github.event.workflow_run.event == 'pull_request'
```

Comment:
This also should not be necessary. In my opinion it is helpful to upload the test results on every run, isn't it?